### PR TITLE
bf: versioning - check if MfaDelete is enabled

### PR DESCRIPTION
**Pull request by nicolas2bert**
If MFA Delete–enabled, we return NotImplemented error.

### CODE REVIEW

File: lib/api/bucketPutVersioning.js [hunk 1]
```diff
@@ -28,18 +28,21 @@ function _parseXML(request, log, cb) {
...
         const validStatuses = ['Enabled', 'Suspended'];
         const validMfaDeletes = [undefined, 'Enabled', 'Disabled'];
         if (validStatuses.indexOf(status) < 0 ||
             validMfaDeletes.indexOf(mfaDelete) < 0) {
             log.debug('illegal versioning configuration');
             return cb(errors.IllegalVersioningConfigurationException);
         }
-        if (mfaDelete) {
+        if (versioningConf && versioningConf.MfaDelete &&
+        versioningConf.MfaDelete[0] && versioningConf.MfaDelete[0] ===
```
Comment (LaurenSpiegel) [inline lib/api/bucketPutVersioning.js]: can't you just do `if (versioningConf && mfaDelete === 'Enabled')`?

Comment (nicolas2bert) [inline lib/api/bucketPutVersioning.js]: oh yes!

**Comment by LaurenSpiegel** [comment]
@ironman-machine go

**Comment by nicolas2bert** [comment]
@ironman-machine try

**Pull request merged**